app/src/cpp/cfc-cpp/jni.cpp

```diff
@@ -162,20 +162,21 @@ namespace {
 
 extern "C" {
 jstring JNICALL
-Java_org_cimbar_camerafilecopy_MainActivity_processImageJNI(JNIEnv *env, jobject instance, jlong matAddr, jstring dataPathObj, jint colorBitsJ)
+Java_org_cimbar_camerafilecopy_MainActivity_processImageJNI(JNIEnv *env, jobject instance, jlong matAddr, jstring dataPathObj, jint modeInt)
 {
 	++_calls;
 
 	// get params from raw address
 	Mat &mat = *(Mat *) matAddr;
 	string dataPath = jstring_to_cppstr(env, dataPathObj);
-	int colorBits = (int)colorBitsJ;
+	int modeVal = (int)modeInt;
+	bool legacyMode = modeVal <= 8; // current scheme: old 4c = 4, old 8c = 8, new = bigger number
```

Comment:
These "modeVal" numbers (4, 8, 68) are CFC only at the moment, and might change in the future. libcimbar *currently* has no such concept, but if we adopt one there we'll probably mirror it here.